tests/baselines/reference/classWithStaticFieldInParameterBindingPattern.2(target=es2015).errors.txt

```diff
@@ -7,7 +7,7 @@ classWithStaticFieldInParameterBindingPattern.2.ts(6,57): error TS2373: Paramete
     class C {}
     (({ [class extends C { static x = 1 }.x]: b = "" }) => { var C; })();
                        ~
-!!! error TS2373: Parameter '{ [class extends C { static x = 1 }.x]: b = "" }' cannot reference identifier 'C' declared after it.
+!!! error TS2373: Parameter 'b' cannot reference identifier 'C' declared after it.

```

Comment:
All of these new error changes don't feel right, no? The problem is that you can't alias the class and the var since it'll be downleveled, but now we're blaming `b` but the error position is still in the old place?

Reply:
I agree that it doesn't completely feel right. I think it's a separate problem though. This is how TS reports errors referring to binding elements in those kinds of positions today:
```ts
// Parameter 'foo' cannot reference identifier 'bar' declared after it.
function test({ foo = bar, b: bar }: any) {}
```

This is a little bit different because this time it's the `.name` that contains the problem... but was the old "parameter span" that much better here? I could bring back the original "parameter span" in those errors but based on the example above I concluded that the new one is fine. Happy to put some extra work into this if you feel this deserves it.

I see what you are saying about downleveling and all but this one errors with all targets. That would make it yet another issue as this (beautiful) piece of code shouldn't be flagged as a problem in es2015+